helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/AvscWriter.java

```diff
@@ -70,19 +117,26 @@ protected void toJson(
                     return;
                 }
                 gen.writeStartObject();
-                gen.writeStringField("type", "enum");
                 avro702Data = writeName(schema, names, contextNamespaceWhenParsed, contextNamespaceWhenParsedUnder702, gen);
-                if (schema.getDoc() != null) {
-                    gen.writeStringField("doc", schema.getDoc());
-                }
+                gen.writeStringField("type", "enum");
                 gen.writeArrayFieldStart("symbols");
                 for (String symbol : schema.getEnumSymbols()) {
                     gen.writeString(symbol);
                 }
                 gen.writeEndArray();
-                writeEnumDefault(schema, gen);
-                writeProps(schema, gen);
-                aliasesToJson(schema, avro702Data.getExtraAliases(), gen);
+                if(retainDefaults) {
+                    writeEnumDefault(schema, gen);
+                }
+                Set<String> claimedPropsEnum = executeAvscWriterPluginsForSchema(schema, gen);
+
+                if(retainNonClaimedProps) {
+                    writeProps(schema, gen, claimedPropsEnum);
+                }
+
+                aliasesToJson(schema, avro702Data.getExtraAliases(), gen, retainSchemaAliases);
```

Comment:
do we have a well defined order in which we print properties? is there one defined in the avro spec we can extend? i think all the "builtins" should come before anything printed by plugins

Reply:
Yes. The order:

1. name
2. namespace
3. type
4. fields
5. symbols
6. items
7. values
8. size
9. default
10. order (for Fields)
11. aliases
12. doc

Since I've implemented the canonical form as an extension of the avsc writer, we are reusing most of its components which includes how we maintain this order.